packages/squiggle-lang/__tests__/library/plot_test.ts

```diff
@@ -29,6 +49,77 @@ describe("Plot", () => {
       })`,
       "Expected a function with one parameter"
     );
+
+    testPlotResult(
+      "default scale based on domain",
+      `Plot.numericFn({
+        fn: {|x: [3, 5]| x * 5}
+      })`,
+      "numericFn",
+      (plot) => {
+        expect(plot.xScale.type).toBe("linear");
+        expect(plot.xScale.min).toBe(3);
+        expect(plot.xScale.max).toBe(5);
+      }
+    );
+
+    testPlotResult(
+      "explicit scale with min/max ignores the domain",
+      `Plot.numericFn({
+        fn: {|x: [3, 5]| x * 5},
+        xScale: Scale.linear({ min: 100, max: 200 })
+      })`,
+      "numericFn",
+      (plot) => {
+        expect(plot.xScale.type).toBe("linear");
+        expect(plot.xScale.min).toBe(100);
+        expect(plot.xScale.max).toBe(200);
+      }
+    );
+
+    testPlotResult(
+      "scale without min/max inherits domain boundaries",
+      `Plot.numericFn({
+        fn: {|x: [3, 5]| x * 5},
+        xScale: Scale.log()
+      })`,
+      "numericFn",
+      (plot) => {
+        expect(plot.xScale.type).toBe("log");
+        expect(plot.xScale.min).toBe(3);
+        expect(plot.xScale.max).toBe(5);
+      }
+    );
+
+    testEvalToMatch(
+      `Plot.numericFn({
+        fn: {|x| x * 5},
+        xScale: Scale.linear({ min: 100 })
+      })`,
+      "'max' and 'min' should either be simultaneously set or unset on xScale for a function plot"
+    );
+
+    testEvalToMatch(
+      `Plot.numericFn({
+        fn: {|x| x * 5},
+        xScale: Scale.linear({ max: 100 })
+      })`,
+      "'max' and 'min' should either be simultaneously set or unset on xScale for a function plot"
+    );
+
+    testPlotResult(
+      "scale with one of min/max inherits domain boundaries",
+      `Plot.numericFn({
+        fn: {|x: [3, 5]| x * 5},
+        xScale: Scale.log({ min: 100 })
```

Comment:
Decided against just throwing an error in these situations? As a user, I'd be a bit surprised that I could do both.

Reply:
I thought there might be some rare cases where this is useful (it's not the same scenario as we discussed on slack), but couldn't think of any, so I made this stricter in the last commit.